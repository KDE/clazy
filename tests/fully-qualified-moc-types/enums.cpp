#include <QtCore/QObject>

namespace NS
{
class TestClass : QObject
{
  Q_OBJECT
    enum E {
    };
    enum class EC {
    };
Q_SIGNALS:
    void function(NS::TestClass::E value); // OK
    void function2(TestClass::E value); // WARN
    void function3(E value); // WARN

    void function(NS::TestClass::EC value); // OK
    void function2(TestClass::EC value); // WARN
    void function3(EC value); // WARN
};
}
