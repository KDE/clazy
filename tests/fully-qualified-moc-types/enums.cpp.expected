fully-qualified-moc-types/enums.cpp:14:20: warning: signal arguments need to be fully-qualified [-Wclazy-fully-qualified-moc-types]
fully-qualified-moc-types/enums.cpp:15:20: warning: signal arguments need to be fully-qualified [-Wclazy-fully-qualified-moc-types]
fully-qualified-moc-types/enums.cpp:18:20: warning: signal arguments need to be fully-qualified [-Wclazy-fully-qualified-moc-types]
fully-qualified-moc-types/enums.cpp:19:20: warning: signal arguments need to be fully-qualified [-Wclazy-fully-qualified-moc-types]
